swarm_dial.go

```diff
@@ -289,7 +289,11 @@ func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
 		the improved rate limiter, while maintaining the outward behaviour
 		that we previously had (halting a dial when we run out of addrs)
 	*/
-	goodAddrs := s.filterKnownUndialables(s.peers.Addrs(p))
+	peerAddrs := s.peers.Addrs(p)
+	if len(peerAddrs) == 0 {
+		return nil, errors.New("no addresses")
```

Comment:
Exported error, please.

Reply:
Appreciate if you can export the "no good addresses" error below too.

Reply:
I just don't see it as an actionable error, and not worth littering the net interface for.

Reply:
Precisely due to the lack of an exported error here, other modules check the peerstore to determine if we have addresses for a peer. That can lead to false positives where we have addresses but none are diallable. Not worth to fix that now, but I did want to colour my reasoning.

Reply:
Maybe we can address this later, when the need arises.